content/en/docs/concepts/services-networking/_index.md

```diff
@@ -5,37 +5,72 @@ description: >
-     without NAT
-   * agents on a node (e.g. system daemons, kubelet) can communicate with all
-     pods on that node
+* containers in same pod can communicate with each other
+* pods can communicate with all other pods on the same or separate [nodes](/docs/concepts/architecture/nodes/)
+without network address translation (NAT)
+* pods can communicate with other pods on same or separate nodes using L2 bridging, native L3 (IPv4, IPv6) networking, or a CNI-specific form of tunnel encapsulation. 
+
+[//]: # (Not sure the agent communications is part of the K8s network model)
+
+[//]: # (Need a picture of a host network as it relates to pods. Do we want to include picture and explanation of kernel network stack? Kernel bypass could be described later)
 
 Note: For those platforms that support `Pods` running in the host network (e.g.
 Linux), when pods are attached to the host network of a node they can still communicate 
 with all pods on all nodes without NAT.
 
-This model is not only less complex overall, but it is principally compatible
-with the desire for Kubernetes to enable low-friction porting of apps from VMs
-to containers.  If your job previously ran in a VM, your VM had an IP and could
-talk to other VMs in your project.  This is the same basic model.
+## Architecture
+
+The Kubernetes network model introduces an architecture that allows you to support your cluster networking requirements.
+
+Figure 1 illustrates the general Kubernetes network architecture.
+
+{{< figure src="/docs/images/k8net-Pod-arch-template.drawio.svg" alt="k8s net arch" class="diagram-large" caption="Figure 1. K8s Network Architecture" >}}
+
+The components of the architecture consist of the following:
+
+* Nodes that can be virtual (VM) or physical.
+* Pods configured on each node with one or more containers.
+* Pods on each node that come with their own IP (v4, v6, dualstack) address and run in their own network namespace.
+* Pods use a virtual "link" between the network namespace pod interface and root network namespace.
+* L2bridge allows communications between pods on the same node or different nodes.
+* CNI plugin supporting different forms of pod networking. Some employ and virtual overlay network in which Pod packets are encapsulated and tunneled between pods on different hosts. Others utilize native network functions. 
+
+Add closing sentence.
+
+### Containers on the same pod
 
 Kubernetes IP addresses exist at the `Pod` scope - containers within a `Pod`
 share their network namespaces - including their IP address and MAC address.
-This means that containers within a `Pod` can all reach each other's ports on
-`localhost`. This also means that containers within a `Pod` must coordinate port
+Containers within a `Pod` can all reach each other's ports on
+`localhost`. 
+
+{{< figure src="/docs/images/k8net-localhost-PodSameHost.drawio.svg" alt="k8s net arch2" class="diagram-large" caption="Figure 2. Container localhost and L2 bridging" >}}
```

Comment:
Watch out for assuming that the network plugin uses layer 2 techniques. A network plugin that works at the IP layer only (eg a Linux `tun` interface plus userland packet encapsulation and decapsulation) is a viable design.

Reply:
https://www.researchgate.net/figure/Kubernetes-Network-routing-to-export-the-services_fig1_337362475 includes an example of a diagram that doesn't rely on OSI layer 2 (although, given it shows Flannel, this is partly down to eliding some detail)

Another option is with text, to highlight that this is _one_ of several viable designs.

Reply:
Understand re: L2-based and other network paradigms. Referenced paper is "researchy" but your point is valid.

Might call for Advanced Topics section later on that covers other paradigms like User-space networking (kernel bypass), Network Service Mesh, maybe eBPF, CNFs, ...